go/cmd/dolt/commands/sql.go

```diff
@@ -948,6 +993,18 @@ func processNonInsertBatchQuery(ctx *sql.Context, se *sqlEngine, query string, s
 		return err
 	}
 
+	foundDoltSQLFunc, err := checkForDoltSQLFunction(sqlStatement)
```

Comment:
This check needs to go in canProcessAsBatchInsert as well. If you did `INSERT INTO my_log (x) select dolt_merge(...)` you would need to flush there.

Also add a bats tests for DOLT_MERGE with non-insert statements

Reply:
Subqueries automatically get sent as nonBatched so I think the solution is to update the checkForDolSQLFunction to look at subqueries as well.